templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2

```diff
@@ -97,16 +121,87 @@ Parameters:
     Default: none
 
 Conditions:
+  2PlusServiceInstances:
+    !Equals [ !Ref ServicesEC2Count, 2 ]
   2PlusWebInstances:
-    !Equals [!Ref WebEC2Count, 2]
+    Fn::Or:
+    - Fn::Equals:
+      - Ref: WebEC2Count
+      - '2'
+    - !Condition 3PlusWebInstances
+  3PlusWebInstances:
+    Fn::Or:
+    - Fn::Equals:
+      - Ref: WebEC2Count
+      - '3'
+    - !Condition 4PlusWebInstances
+  4PlusWebInstances:
+    Fn::Or:
+    - Fn::Equals:
+      - Ref: WebEC2Count
+      - '4'
+    - !Condition 5PlusWebInstances
+  5PlusWebInstances:
+    Fn::Or:
+    - Fn::Equals:
+      - Ref: WebEC2Count
+      - '5'
+    - !Condition 6PlusWebInstances
+  6PlusWebInstances:
+    Fn::Equals:
+    - Ref: WebEC2Count
+    - '6'
 
-  BatchLayerNeeded:
+  BatchLayerNeeded: #batch is background
     !Not [!Equals [!Ref BatchEC2Size, none]]
+  1PlusBatchInstances:
+    Fn::Or:
```

Comment:
I worked out why the logic works this way and I was sad :(

CloudFormation conditional functions are a bit limited.

But perhaps Jinja could generate these blocks more efficiently?

Reply:
maybe, after working out how to build them, its a lot easier, much better than original vpc style